server/drivers/crate/index.js

```diff
@@ -69,6 +67,26 @@ async function runQuery(query, connection) {
   }
 }
 
+/**
+ * Get connection string with options
+ * @param {*} connection
+ */
+function getConnectionString(connection) {
+  let connectionString = connection.ssl ? 'https://' : 'http://';
+  if (connection.username !== undefined && connection.username.length !== 0) {
+    connectionString += connection.username;
+    if (connection.password !== undefined && connection.password.length !== 0)
+      connectionString += ':' + connection.password;
+    connectionString += '@';
+  }
+  connectionString += connection.host + ':';
+  connectionString +=
+    connection.port === undefined || connection.port.length === 0
```

Comment:
similar challenge here as well, but we should handle string or number for port. For this I'd go with 

```js
connectionString += Boolean(connection.port) ? CRATE_DEFAULTPORT : connection.port;
```

Reply:
the key `port` is defined as text, shouldn't I expect to only handle text?
changed it to `if (connection.port)`
Also I removed the default port, after checking `node-crate` again